### PR TITLE
RFE: update genfscon documentation

### PR DESCRIPTION
**Pull request by cgzones**
Partial paths can be used generally for virtual kernel filesystems, e.g. cgroup2, securityfs, selinuxfs, and not just for the proc filesystem.

The genfscon statement supports an optional filetype specifier.

### CODE REVIEW

**Review by stephensmalley — COMMENTED** [review]
The situation is somewhat more complex than described here. See https://github.com/SELinuxProject/selinux-kernel/issues/2 for discussion of safety and use of genfscon. The kernel mechanism itself does not limit use of genfscon to only these cases; one can for example use it to label an initramfs-based rootfs, or a squashfs or even technically ext4 (but the latter wouldn't be safe).

**Comment by jbrindle** [comment]
@cgzones Any chance you can publish an update to this based on Stephen's comment above, or would you prefer us to pick it up?

**Comment by cgzones** [comment]
Updated the wording a bit, feel free to suggest improvements.

**Review by stephensmalley — APPROVED** [review]

**Comment by stephensmalley** [comment]
Reviewed-by: Stephen Smalley <stephen.smalley.work@gmail.com>

File: src/file_labeling_statements.md [hunk 1]
```diff
@@ -241,14 +257,20 @@ genfscon selinuxfs / system_u:object_r:security_t:s0
 ```
 
 ```
-# The following show some example /proc entries. Note that the
-# /kmsg has the highest sensitivity level assigned (s15) because
+# The following examples show pseudo kernel filesystem entries. Note that
+# the /kmsg has the highest sensitivity level assigned (s15) because
 # it is a trusted process.
```
Comment (jbrindle) [inline src/file_labeling_statements.md]: nit, it's a file, not process. It is s15 because it could contain data from any level on the system

**Review by jbrindle — CHANGES_REQUESTED** [review]

File: src/file_labeling_statements.md (hunk 1 above)
Comment (cgzones) [reply src/file_labeling_statements.md]: adjusted

**Comment by jbrindle** [comment]
reviewed-by: Joshua Brindle <brindle@gmail.com>

**Pull request merged**